packages/components/src/slot-fill/bubbles-virtually/use-slot.js

```diff
@@ -11,10 +16,8 @@ import SlotFillContext from './slot-fill-context';
 
 export default function useSlot( name ) {
 	const registry = useContext( SlotFillContext );
-
-	const slot = registry.slots[ name ] || {};
-	const slotFills = registry.fills[ name ];
-	const fills = useMemo( () => slotFills || [], [ slotFills ] );
+	const slots = useSnapshot( registry.slots );
+	const slot = slots.get( name );
```

Comment:
The important bit here is that, this hook only causes a re-render if the slot with the given name change, not any slot.

Reply:
Same